File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.kt
class Solution {
    fun containsNearbyDuplicate(nums: IntArray, k: Int): Boolean {
        val hm = HashMap<Int, Int>()
        for (i in nums.indices) {
            if (hm.containsKey(nums[i])) {
                if (hm[nums[i]!!]!! >= (i-k)) return true
            }
            hm[nums[i]] = i
        }
        return false
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        N = len(nums)
        d = {}
        for i in range(N):
            key = nums[i]
            if key in d.keys():
                if i - d[key] <= k:
                    return True
            d[key] = i
        return False
    
        